source/extensions/filters/http/aws_lambda/sts_credentials_provider.cc

```diff
@@ -152,17 +166,15 @@ StsConnectionPool::Context *StsCredentialsProviderImpl::find(
   }else{
      // since there is no existing connection pool then we need to create one
      conn_pool =  connection_pools_
-          .emplace(role_arn, conn_pool_factory_->build(
+          .emplace(role_arn_lookup, conn_pool_factory_->build(
               role_arn, this, StsFetcher::create(cm_, api_)))
           .first;
   }
 
-  // Short circuit any additional checks if we are the base arn
-  // Use getInteger as it doesnt check against default runtime and
-  // So that runtime enablement is not decided at per filter runtime
-  // via a proto based flag. 
-  if (role_arn == default_role_arn_  || Runtime::getInteger(
```

Comment:
Are we dropping support for the runtime val?

Reply:
Runtime dropped support for us

Reply:
hahahahahaha, were we ever using it?

Reply:
Yeah 1.11s version worked its just in envoy 1.23 that this became locked down

Reply:
```
uint64_t getInteger(absl::string_view feature, uint64_t default_value) {
  ASSERT(absl::StartsWith(feature, "envoy."));
  if (Runtime::LoaderSingleton::getExisting()) {
    return Runtime::LoaderSingleton::getExisting()->threadsafeSnapshot()->getInteger(
        std::string(feature), default_value);
  }
  ENVOY_LOG_TO_LOGGER(Envoy::Logger::Registry::getLog(Envoy::Logger::Id::runtime), debug,
                      "Unable to use runtime singleton for feature {}", feature);
  return default_value;
}
```

Became 

`uint64_t getInteger(absl::string_view feature, uint64_t default_value) {
  if (absl::StartsWith(feature, "envoy.")) {
    // DO NOT ADD MORE FLAGS HERE. This function deprecated.
    if (feature == "envoy.http.headermap.lazy_map_min_size") {
      return absl::GetFlag(FLAGS_envoy_headermap_lazy_map_min_size);
    }
  }
  if (absl::StartsWith(feature, "re2.")) {
    if (feature == "re2.max_program_size.error_level") {
      return absl::GetFlag(FLAGS_re2_max_program_size_error_level);
    } else if (feature == "re2.max_program_size.warn_level") {
      return absl::GetFlag(FLAGS_re2_max_program_size_warn_level);
    }
  }
  IS_ENVOY_BUG(absl::StrCat("requested an unsupported integer ", feature));
  return default_value;
}`
Which was part of the upstream lockdown on enums and emitting ENVOY_BUG